src/main/java/de/danielbechler/diff/access/PropertyAwareAccessor.java

```diff
@@ -29,4 +29,8 @@
 	Set<Annotation> getReadMethodAnnotations();
 
 	<T extends Annotation> T getReadMethodAnnotation(Class<T> annotationClass);
+
+   <T extends Annotation> T getAnnotation(Class<T> annotationClass);
+
+   int getFieldModifiers();
```

Comment:
If you don't feel comfortable adding new configuration options, maybe the easiest solution would be to just replace your `CibetInclusionResolver` with a custom `Introspector`. The good thing about that is that it's easy to do, very flexible and already fully supported. Your `CibetIntrospector` could even remain in your code base.

Your way to exclude properties seems to be very specific for your use case. For example, you are assuming that the field name corresponds to the getter and setter name (even when the field is private). While this may be desirable in some cases (like yours), I don't think it's a good default, because it involves a lot of unexpected magic. Point in case: what happens when there are fields `private int _amount` and `private transient int amount` and `getAmount()` actually returns `_amount`?

I think a custom `Introspector` would most definitely give you all the freedom you need, without the need to expose any more "internals" via `DiffNode`. Do you think that could work for you?